src/cpp/session/modules/SessionDataImportV2.R

```diff
@@ -255,29 +259,51 @@
       cacheDataCode <- list()
 
       localFile <- unlist(options$localFiles[[resource]])
+      canCacheData <- identical(options$canCacheData, TRUE)
+      cacheDataWorkingDir <- !identical(options$cacheDataWorkingDir, NULL) && options$cacheDataWorkingDir == TRUE
 
-      if (importFromUrl)
+      if (importFromUrl && canCacheData)
       {
-         if (identical(localFile, NULL))
+         cacheVariableName <- options$cacheVariableNames[[resource]]
+
+         if (cacheDataWorkingDir)
          {
-            localFile <- tempfile(
-               tmpdir = dirname(tempdir()),
-               fileext = functionInfo$cacheFileExtension
+            cacheDataCode <- append(
+               cacheDataCode,
+               paste(
+                  cacheVariableName,
+                  " <- tempfile(tmpdir = getwd(), fileext = \"",
```

Comment:
yes, that is better, changed. Generated code for XLS imported from a URL looks like this with this change:

```
library(readxl)
fy2015f_4050_final_file <- file.path(getwd(), "fy2015f_4050_final.xls")
download.file("http://www.huduser.org/portal/datasets/fmr/fmr2015f/FY2015F_4050_Final.xls", fy2015f_4050_final_file)
fy2015f_4050_final <- read_excel(fy2015f_4050_final_file)
View(fy2015f_4050_final)
```